vcf.c

```diff
@@ -645,6 +650,35 @@ int bcf_hdr_parse(bcf_hdr_t *hdr, char *htxt)
         needs_sync += bcf_hdr_add_hrec(hdr, hrec);
         p += len;
     }
+
+    // Detect incorrect header lines, print a warning, and skip them.
+    // Alternatively, the program could refuse to work with such VCF. However,
+    // many VCF operations do not really care about a few malformed header
+    // lines.
+    if ( strncmp("#CHROM\tPOS",p,10) )
```

Comment:
This is actually due to  `bcf_hdr_add_sample()` punishing invalid input by calling `abort()`.  Your `##CHROM` line included a trailing tab, which caused `bcf_hdr_add_sample()` to print:

```
[E::bcf_hdr_add_sample] Empty sample name: trailing spaces/tabs in the header line?
```
and then abort().

I think it should be fixable to return non-zero instead without getting in to too deep a hole.